tests/operator/operator.go

```diff
@@ -3285,3 +3085,226 @@ func getVirtLauncherSha(deploymentConfigStr string) (string, error) {
 
 	return config.VirtLauncherSha, nil
 }
+
+func deleteAllKvAndWait(ignoreOriginal bool, originalKvName string) {
+	GinkgoHelper()
+
+	virtClient := kubevirt.Client()
+	Eventually(func(g Gomega) {
+		kvs := libkubevirt.GetKvList(virtClient)
+
+		deleteCount := 0
+		for _, kv := range kvs {
+			if ignoreOriginal && kv.Name == originalKvName {
+				continue
+			}
+			deleteCount++
+			if kv.DeletionTimestamp == nil {
+				GinkgoLogr.Info("deleting the kv object", "namespace", kv.Namespace, "name", kv.Name)
+				g.Expect(
+					virtClient.KubeVirt(kv.Namespace).Delete(context.Background(), kv.Name, metav1.DeleteOptions{}),
+				).To(Succeed())
+			}
+		}
+
+		g.Expect(deleteCount).To(BeZero(), "still waiting on %d kvs to delete", deleteCount)
+	}).WithTimeout(240 * time.Second).WithPolling(1 * time.Second).Should(Succeed())
+}
+
+func ensureShasums() error {
```

Comment:
Would it make sense to keep the Expects in this helper (and others) instead of returning `error`? Not sure what is more favorable.

Reply:
I really think that gomega assertions in a helper function is bad practice, mostly because it's really hard to get the actual error location from gomega error message

Reply:
I asked about have it the other way around  (no expect in helpers) :laughing: 

Using expect with helpers has some drawbacks, you need to put attention to the assertion offset (ExpectWithOffset), writing helper with not depending on Expect make it reusable and easier to port (in case its needed).
Most importantly, when you read the test code there are no surprises, no need to dig in the function and find some asserts: 
shouldDoThisAndThat()
vs.
Expect(shouldDoThisAndThat).To(Succeed())

Reply:
I would add it to our code convention doc

Reply:
In general I agree, but as you can see above, for `Eventually` it isn't that easy. I'd say it's hard to make this a general convention.

Reply:
yes and know... I wouldn't change it, but in theory, we can use another retry solution, like the `wait` package. But having two retry solutions seems strange.